hibernate-core/src/main/java/org/hibernate/Hibernate.java

```diff
@@ -109,81 +110,80 @@ public static boolean isInitialized(Object proxy) {
 		}
 		else if ( proxy instanceof PersistentAttributeInterceptable ) {
 			final PersistentAttributeInterceptor interceptor = ( (PersistentAttributeInterceptable) proxy ).$$_hibernate_getInterceptor();
-			return !(interceptor instanceof EnhancementAsProxyLazinessInterceptor);
+			return !( interceptor instanceof EnhancementAsProxyLazinessInterceptor );
 		}
 		else if ( proxy instanceof PersistentCollection ) {
 			return ( (PersistentCollection<?>) proxy ).wasInitialized();
 		}
+		else if ( proxy instanceof LazyInitializable ) {
+			return ( (LazyInitializable) proxy ).wasInitialized();
+		}
```

Comment:
You don't need two `if` branches, as `PersistentCollection` extends `LazyInitializable`:

```suggestion
		else if ( proxy instanceof LazyInitializable ) {
			return ( (LazyInitializable) proxy ).wasInitialized();
		}
```